src/components/Tips.js

```diff
@@ -0,0 +1,118 @@
+import { h } from 'preact'
+import { useRef, useEffect } from 'preact/hooks'
+import { useNavigation, useI18n, useSoftkey, usePopup } from 'hooks'
+import { ListView } from 'components'
+import { goto } from 'utils'
+
+export const Tips = () => {
+  const containerRef = useRef()
+  const listRef = useRef()
+  const i18n = useI18n()
+  const [, setNavigation, getCurrent] = useNavigation('Tips', containerRef, listRef, 'y')
+
+  const [showReadPopup] = usePopup(ReadPopup, { mode: 'fullscreen' })
+  const [showSearchPopup] = usePopup(SearchPopup, { mode: 'fullscreen' })
+  const [showSwitchPopup] = usePopup(SwitchPopup, { mode: 'fullscreen' })
+  const [showAboutWikipediaPopup] = usePopup(AboutPopup, { mode: 'fullscreen' })
+
+  const onKeyCenter = () => {
+    const { index } = getCurrent()
+    const item = items[index]
+    item.action()
+  }
+
+  const onReadPopupSelected = () => {
+    showReadPopup({ onSearchPopupSelected })
+  }
+
+  const onSearchPopupSelected = () => {
+    showSearchPopup({ onSwitchPopupSelected })
+  }
+
+  const onSwitchPopupSelected = () => {
+    showSwitchPopup({ showAboutWikipediaPopup })
+  }
+
+  useSoftkey('Tips', {
+    left: i18n('softkey-back'),
+    onKeyLeft: () => history.back(),
+    center: i18n('centerkey-select'),
+    onKeyCenter,
+    onKeyBackspace: () => history.back()
+  })
+
+  useEffect(() => {
+    setNavigation(0)
+  }, [])
+
+  const items = [
+    { title: i18n('tips-read'), action: onReadPopupSelected },
+    { title: i18n('tips-search'), action: onSearchPopupSelected },
+    { title: i18n('tips-switch'), action: onSwitchPopupSelected },
+    { title: i18n('tips-about'), action: showAboutWikipediaPopup }
+  ]
+
+  return (
+    <div class='tips' ref={containerRef}>
+      <ListView
+        header={i18n('tips-header')}
+        items={items}
+        containerRef={listRef}
+      />
+    </div>
+  )
+}
+
+const tip = (origin, key, close, onNext, onTry) => {
+  const i18n = useI18n()
+  const aboutTip = key === 'about'
+
+  const onTryHandler = () => {
+    if (key === 'search') {
+      close()
+      onTry()
+    } else {
+      onTry(close)
+    }
+  }
+
+  const getBackgroundStyle = () => {
+    return { backgroundImage: aboutTip ? 'url(/images/onboarding-0.png)' : `url(images/tip-${key}-animation.gif` }
+  }
+
+  useSoftkey(origin, {
+    left: i18n('softkey-back'),
+    onKeyLeft: close,
+    center: !aboutTip ? i18n('softkey-try') : '',
+    onKeyCenter: !aboutTip ? onTryHandler : null,
+    right: !aboutTip ? i18n('softkey-next') : '',
+    onKeyRight: !aboutTip ? () => { close(); onNext() } : null,
+    onKeyBackspace: close
+  }, [])
+
+  return (
+    <div class={'tip'}>
+      <div class='tip-media'>
+        <div class={`tip-${aboutTip ? 'image' : 'animation'}`} style={getBackgroundStyle()} />
```

Comment:
Building message keys like that is generally discouraged because it makes them unsearchable. For instance, if you're wondering if `tip-image` is still used anywhere, you search for it but you won't find it. One exception is for messages ending with a number (step-1, step-2, etc), it is somewhat common knowledge to ignore the number when searching for it.

An alternative that is not too much longer is `${aboutTip ? 'tip-image' : 'tip-animation'}`

For cases like `tip-${key}-message`, you can let individual tips provide their own set of messages.

Another strategy you might see is adding a comment
// this line can generate the following messages
// tip-image
// tip-animation
It works when you search but is often more verbose than needed and can get out of sync with the code.

Reply:
Yea you're right! I've had that searching issue happen to me before

> An alternative that is not too much longer is ${aboutTip ? 'tip-image' : 'tip-animation'}

I like this idea